google/auth/external_account.py

```diff
@@ -425,6 +484,23 @@ def _initialize_impersonated_credentials(self):
             ),
         )
 
+    def _token_info_username(self):
+        if not self._token_info_introspection:
+            try:
+                self.token_info_introspection(None)
+            except exceptions.GoogleAuthError:
+                # A failure of introspection shouldn't break the flow.
+                pass
+
+        # One runnig of token_info_introspection() is not guaranteed we have a
+        # valid inspection result cached. A failed introspection may results
+        # the cache unchanged and an empty cache still there.
+        return (
+            None
+            if not self._token_info_introspection
+            else self._token_info_introspection.get("username")
+        )
```

Comment:
```suggestion
        return self._token_info_introspection.get("username", None)
```

Assuming this is just a regular dict

Reply:
`self._token_info_introspection` may remain `None` and won't have a `get` method.

Reply:
Gotcha. Why use None instead of an empty dict? Do you need to make any decisions based on it being an optional type?

Reply:
Good raise. Seems reasonable to initialize it as an empty dict.